int32 encoder1_count
int32 encoder2_count